src/axom/inlet/VerifiableScalar.hpp

```diff
@@ -78,11 +89,21 @@ class VerifiableScalar
    * \brief Registers the function object that will verify this object's contents
    * during the verification stage.
    * 
-   * \param [in] The function object.
+   * \param [in] lambda The function object.
+   *****************************************************************************
+  */
+  VerifiableScalar& registerVerifier(
+    std::function<bool(const axom::inlet::Field&)> lambda);
+
+  /*!
+   *****************************************************************************
+   * \brief Registers the function object that will verify this object's contents
+   * during the verification stage.
+   *
+   * \param [in] verifier The function object.
    *****************************************************************************
   */
-  virtual VerifiableScalar& registerVerifier(
-    std::function<bool(const axom::inlet::Field&)> lambda) = 0;
+  virtual VerifiableScalar& registerVerifier(Verifier verifier) = 0;
```

Comment:
I may be in the minority, but using a type alias for a user-facing function makes it hard to know what the verifiers function signature should be. @kennyweiss / @joshessman-llnl  thoughts?

Reply:
I enabled the docs here: https://axom.readthedocs.io/en/feature-pauli2-custom-verifier-log-to-vector/

The type alias way seems reasonable to me from the doxygen, e.g.:
* Field: https://axom.readthedocs.io/en/feature-pauli2-custom-verifier-log-to-vector/doxygen/html/classaxom_1_1inlet_1_1Field.html
* Function: https://axom.readthedocs.io/en/feature-pauli2-custom-verifier-log-to-vector/doxygen/html/classaxom_1_1inlet_1_1Function.html

Reply:
For clarify, we might want to group the `registerVerifier` functions together in the docs, 
e.g. see `inlet::Field` (https://axom.readthedocs.io/en/feature-pauli2-custom-verifier-log-to-vector/doxygen/html/classaxom_1_1inlet_1_1Field.html)

I think something like following should help:
```cpp
//@{
//!  @name Functions for registering and invoking verifiers

<all code related to `registerVerifier` here>
<all code related to `verify` here>

//@}

Reply:
`verify()` doesn't just call the custom verifiers -- it also validates the calls to other functions like `required()`, `range()`, `validValues()`. `registerVerifier()` is just an extra flavor of this. Given that, though, there are some like `defaultValue()` which are a little different.

An additional complications is that functions from the parent class (e.g. a few `get()` instances and a `registerVerifier()` for Field) get generated at the bottom of the class instead.

Give the above, I'd like to suggest that this is outside the scope of this PR. If a bigger rearrangement of this documentation is desired, it should be handled separately as it would touch a lot of lines.